src/m3ninx/index/segment/fst/writer.go

```diff
@@ -198,27 +236,32 @@ func (w *writer) WritePostingsOffsets(iow io.Writer) error {
 			return err
 		}
 
+		fieldsPL := roaring.NewPostingsList()
```

Comment:
nit: You could reuse a postings list with cached containers here.  Which for the compaction code path, since it happens basically every time you add any documents, it might very well be necessary.

See how `currPostingsList` is constructed and used in the `termsIterFromSegments` here (notice built from a bitmap that is using pooling, then Reset is called at top of the inner loop).  This made a large difference in allocations:
https://github.com/m3db/m3/blob/master/src/m3ninx/index/segment/builder/multi_segments_terms_iter.go

Reply:
Ah nice! I didn't do this cause upstream Pilosa's reset doesn't do container reuse smartly, didn't realise we had fork of pilosa. Will update.

Reply:
done